core/src/providers/openai.rs

```diff
@@ -1415,17 +1415,23 @@ impl LLM for OpenAILLM {
     }
 
     async fn initialize(&mut self, credentials: Credentials) -> Result<()> {
-        match credentials.get("OPENAI_API_KEY") {
-            Some(api_key) => {
-                self.api_key = Some(api_key.clone());
+        // Give priority to `CORE_DATA_SOURCES_OPENAI_API_KEY` env variable
```

Comment:
same comment wrt asyncness of the method

Reply:
Yes but that's a broader refactor as this method is part of the signature of the trait above